src/ProxyClient/Cloudflare.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCache package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCache\ProxyClient;
+
+use FOS\HttpCache\ProxyClient\Invalidation\ClearCapable;
+use FOS\HttpCache\ProxyClient\Invalidation\PurgeCapable;
+use FOS\HttpCache\ProxyClient\Invalidation\TagCapable;
+
+/**
+ * Cloudflare HTTP cache invalidator.
+ *
+ * Additional constructor options:
+ * - zone_identifier       Identifier for your Cloudflare zone you want to purge the cache for
+ * - authentication_token  API authorization token, requires Zone.Cache Purge permissions
+ *
+ * @author Simon Jones <simon@studio24.net>
+ */
+class Cloudflare extends HttpProxyClient implements ClearCapable, PurgeCapable, TagCapable
+{
+    /**
+     * @see https://api.cloudflare.com/#getting-started-endpoints
+     */
+    private const API_ENDPOINT = 'https://api.cloudflare.com/client/v4';
+
+    /**
+     * Batch URL purge limit.
+     *
+     * @see https://api.cloudflare.com/#zone-purge-files-by-url
+     */
+    private const URL_BATCH_PURGE_LIMIT = 30;
+
+    /**
+     * Array of data to send to Cloudflare for purge by URLs request.
+     *
+     * @var array
+     */
+    private $purgeByUrlsData = [];
+
+    /**
+     * JSON encode data.
+     *
+     * @param $data
+     *
+     * @return false|string
+     */
+    protected function encode($data)
+    {
+        return json_encode($data, JSON_UNESCAPED_SLASHES);
```

Comment:
Would it be better to throw exception here instead of `false` on encoding failure?
Otherwise `false` will be sent as body of invalidation request?

Reply:
That's a good idea, but `JSON_THROW_ON_ERROR` is only available since PHP 7.3 so is that OK? This library is currently marked as PHP 7.2+ in the composer.json file. An alternative is to detect a false return value and manually throw an exception.

Reply:
Personally I would go with return value check, something like that

```
$json = json_encode($data);
if (false === $json && json_last_error() !== JSON_ERROR_NONE) { // Technically only of those checks is needed.
    throw new \InvalidArgumentException('Cannot encode "$data": ' . json_last_error_msg());
}
```

Reply:
AFAIK json_encode always returns false on error, so I've just used that check.